sale_require_ref/data/sale_exception_data.xml

```diff
@@ -1,17 +1,13 @@
-<?xml version="1.0" encoding="utf-8"?>
-<odoo>
-  <data noupdate="1">
-
-    <record id="require_ref" model="sale.exception">
+<?xml version="1.0"?>
+<odoo noupdate="1">
+    <record id="require_ref" model="exception.rule">
```

Comment:
now that you change the model from sale.exception to exception.rule please update again the file name to exception_rule_data.xml